### PR TITLE
[Feature] Automatic full statistics collection supports running in a specific time period

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticAutoCollector.java [hunk 1]
```diff
@@ -51,6 +56,11 @@ protected void runAfterCatalogReady() {
             return;
         }
 
+        if (!checkoutAnalyzeTime(Instant.ofEpochMilli(System.currentTimeMillis())
+                .atZone(TimeUtils.getTimeZone().toZoneId()).toLocalTime())) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticAutoCollector.java:60]: `LocalTime.now(TimeUtils.getTimeZone().toZoneId())`?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|11|13|84.62%|[61, 119]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticAutoCollector.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticAutoCollector.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 15 (86.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|10|12|83.33%|[59, 117]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**